ros/src/waypoint_updater/waypnt_updater.py

```diff
@@ -586,25 +589,25 @@ def distance_lambda(a, b): return math.sqrt(
             (a.x-b.x)**2 + (a.y-b.y)**2)
         # TODO: move away from using final waypoint, just use waypoints
         # since we have saved original v info within the structure
-        if self.final_waypoints:
-            dist = distance_lambda(self.final_waypoints[0].get_position(),
+        if self.waypoints and self.last_search_distance:
+            dist = distance_lambda(
+                self.waypoints[self.final_waypoints_start_ptr-1].get_position(),
```

Comment:
I was worried about negative indexing if self.final_waypoints_start_ptr=0, but remembered that Python indexing is awesome!  No problem.